src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+: Compares itself against other `flex` properties before determining 
+what fraction of the total extra space each `Flexible` widget receives.
+
+`fit` 
+: Determines whether a `Flexible` widget fills all of its extra space.
+
+#### Example: Testing flex values
+{:.no_toc}
+{{site.alert.secondary}}
+  In the following example, two `Flexible` widgets wrap two `BlueBox` widgets. 
+  The `Flexible` widgets contain `flex` properties with values set to 1 (the default value).
+
+  When `flex` properties are compared, 
+  the ratio between the values determines 
+  what fraction of the total extra space each `Flexible` widget receives.
+
+  In this example, the sum of the `flex` values (2),
+  determines that each `Flexible` widget increases
+  by half of the free space. 
+  The fixed-width widget remains the same size.
+
+  **1.** Click the **Run** button. 
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/82e4dd24028034ae03ba0ddc71bf59e5
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=82e4dd24028034ae03ba0ddc71bf59e5&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+{{site.alert.tip}}
+  Before moving to the next code example, change the `flex` properties to other values, 
+  such as 2 and 1.
+{{site.alert.end}}
+
+#### Example: Testing fit properties
+{:.no_toc}  
+{{site.alert.secondary}}
+  The following example displays the `fit` property,
+  which can have one of two values:
+
+  `FlexFit.loose`
+  : The widget's preferred size is used. (Default)
+
+  `FlexFit.tight`
+  : Forces the widget to fill all of its extra space. 
+
+  In this example, modify the `fit` properties to
+  make the `Flexible` widgets fill the free space.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Change the `fit` values to `FlexFit.tight`, 
+  and run again.   
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/ba0f40356d1023066d960f6de2be1a4b
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=ba0f40356d1023066d960f6de2be1a4b&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Expanded widget
+
+Similar to the `Flexible` widget, the `Expanded` widget can wrap a widget and force the widget to fill extra space. 
+
+{{site.alert.tip}}
+  **What's the difference between Flexible and Expanded?**
+    Use `Flexible` to resize widgets in a `Row` or `Column`,
+    so you can adjust child widgets' spacing while keeping the
+    relation with their parent widget.
+    `Expanded` changes the constraints of the child widgets
+    to fill any empty space.
+{{site.alert.end}} 
+
+#### Example: Filling extra space
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example demonstrates how the `Expanded` widget forces its child widget to fill extra space.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Wrap the second `BlueBox` widget in an `Expanded` widget. 
+
+  For example: 
+
+  ```dart
+  Expanded(child: BlueBox(),),
```

Comment:
It's kind of weird to have the `,)` when it's all on one line. I'd expect no comma before that paren.